### DISCUSSION TITLE
How can you use external modules with a local web viewer?

### DISCUSSION THREAD

**Discussion by napter** [jscad/OpenJSCAD.org#957] [Q&A] [answered]
I am creating a jscad script that imports another script. I would like to test this visually to see how it renders, so I downloaded the web viewer and tried to load it. However the browser can't find the script I'm importing and gives the error "Uncaught Error: Cannot find relative path to module". I assume if I import third party npm modules I will get that error as well.

Is there a way to test this in the web viewer?
Is there a better way for me to test and get a live preview? 

Thanks!

**Comment by z3dev** [comment, marked as answer]
@napter thanks for using JSCAD.

So, from what you explained, you’ll want to create a project directory, and put all your components inside the directory. Then drag and drop the project directory to the webpage.

there are some example projects in the examples. See the module-design

https://github.com/jscad/OpenJSCAD.org/tree/master/packages/examples
